app/screens/integration_selector/integration_selector.tsx

```diff
@@ -192,40 +187,51 @@ function IntegrationSelector(
         }
 
         const itemKey = extractItemKey(dataSource, item);
-        const currentSelected: Dictionary<UserProfile> | Dictionary<DialogOption> | Dictionary<Channel> = multiselectSelected;
-        const multiselectSelectedItems = {...currentSelected};
 
         switch (dataSource) {
             case ViewConstants.DATA_SOURCE_USERS: {
-                if (currentSelected[itemKey]) {
-                    delete multiselectSelectedItems[itemKey];
-                } else {
-                    multiselectSelectedItems[itemKey] = item as UserProfile;
-                }
+                setMultiselectSelected((current) => {
```

Comment:
This can be extra simplified by doing something like this:

```
// outside of the component
const toggleFromMap = <T extends DialogOption | Channel | UserProfile>(current: MultiselectSelectedMap, key: string, item: T) => {
    const newMap = {...current};
    if (current[key]) {
        delete newMap[key];
    } else {
        newMap[key] = item;
    }

    return newMap;
}

// here
setMultiselectSelected((cur) => {
    return toggleFromMap(cur, itemKey, item as UserProfile);
})
```

Probably a similar approach can be taken in the other similar functions.

Reply:
I'll do it! 👍

Reply:
https://github.com/mattermost/mattermost-mobile/pull/6791

Reply:
I'll check generics too, I don't remember using them ever